### PR TITLE
Fix registry secret env vars

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
This PR does the following:
- Fetches the BSL secret for AWS providers
- Parses the secret and updates the registry env vars accordingly (only AWS)
- Updates registry test cases

### CODE REVIEW

File: controllers/registry.go [hunk 1]
```diff
@@ -270,30 +284,54 @@ func (r *VeleroReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLoc
...
+
+	r.Log.Info(fmt.Sprintf("Fetched secret: %#v", secret))
+	// parse the secret and get aws access_key and aws secret_key
+	AWSAccessKey, AWSSecretKey, err := r.parseAWSSecret(secret, secretKey)
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation %s/%s", secretName, bsl.Namespace, bsl.Name))
+		return nil, err
+	}
+
+	r.Log.Info(fmt.Sprintf("AWSAccessKey: %s, AWSSecretKey: %s", AWSAccessKey, AWSSecretKey))
```
Comment (dymurray) [inline controllers/registry.go:330]: We don't want to print the credentials I don't think. Technically they are sensitive

Comment (shubham-pampattiwar) [reply controllers/registry.go]: Thanks @dymurray , this was an unintentional error on my end, (those were my debug logs :) )
Removed all the inappropriate logs.

**Review by dymurray — APPROVED** [review]

File: controllers/registry.go [hunk 2]
```diff
@@ -355,3 +390,66 @@ func (r *VeleroReconciler) getGCPRegistryEnvVars(bsl *velerov1.BackupStorageLoca
...
+		// check for secret key
+		matchedSecretKey, err := regexp.MatchString("\\baws_secret_access_key\\b", line)
+		if err != nil {
+			r.Log.Info("Error finding secret access key for the supplied AWS credential")
+			return AWSAccessKey, AWSSecretKey, err
+		}
+
+		if matchedSecretKey {
+			cleanedLine := strings.ReplaceAll(line, " ", "")
+			AWSSecretKey = strings.Split(cleanedLine, "=")[1]
```
Comment (dymurray) [inline controllers/registry.go:450]: We need to check that there are 2 elements in this array I think

File: controllers/registry.go [hunk 3]
```diff
@@ -355,3 +390,66 @@ func (r *VeleroReconciler) getGCPRegistryEnvVars(bsl *velerov1.BackupStorageLoca
...
+		matchedAccessKey, err := regexp.MatchString("\\baws_access_key_id\\b", line)
+
+		if err != nil {
+			r.Log.Info("Error finding access key id for the supplied AWS credential")
+			return AWSAccessKey, AWSSecretKey, err
+		}
+
+		if matchedAccessKey {
+			cleanedLine := strings.ReplaceAll(line, " ", "")
+			AWSAccessKey = strings.Split(cleanedLine, "=")[1]
```
Comment (dymurray) [inline controllers/registry.go:438]: same here

**Pull request merged**